storage/rocksdb/rdb_datadic.cc

```diff
@@ -2117,10 +2118,11 @@ int Rdb_key_def::unpack_utf8_str(
   uchar *const dst_end = dst + field->pack_length();
 
   while (src < src_end) {
-    my_wc_t wc = (src[0] << 8) | src[1];
-    src += 2;
+    my_wc_t wc = (bytes == 3) ? (src[0] << 16) | (src[1] << 8) | src[2]
```

Comment:
Nit - This is repeated at 3 places, should we expose a small helper method around this.

Reply:
It can be done but template instantiation should create separate methods for `bytes == 3` and `bytes != 3`.